pkg/did/wellknown/wellknown.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package wellknown contains a function for validating a DID's service against its well-known DID configuration.
+package wellknown
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/client/didconfig"
+	diddoc "github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+
+	"github.com/trustbloc/wallet-sdk/pkg/api"
+	"github.com/trustbloc/wallet-sdk/pkg/common"
+)
+
+const linkedDomainsServiceType = "LinkedDomains"
+
+// ValidateLinkedDomains validates the given DID's Linked Domains service against its well-known DID configuration.
+// It returns a bool indicating whether it's valid and the service's URL.
+// The DID document associated with the given DID must specify only a single service.
+// If there are multiple URLs for a given service, only the first will be checked.
+// The HTTP client parameter is optional. If not provided, then a default client will be used.
+func ValidateLinkedDomains(did string, resolver api.DIDResolver,
+	httpClient didconfig.HTTPClient,
+) (bool, string, error) {
+	if resolver == nil {
+		return false, "", errors.New("no resolver provided")
+	}
+
+	if httpClient == nil {
+		httpClient = common.DefaultHTTPClient()
+	}
+
+	didDocResolution, err := resolver.Resolve(did)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to resolve DID: %w", err)
+	}
+
+	if len(didDocResolution.DIDDocument.Service) == 0 {
+		return false, "", errors.New("resolved DID document has no services specified")
+	} else if len(didDocResolution.DIDDocument.Service) > 1 {
```

Comment:
This condition is not valid. The DID Document can have more than 1 services. For LinkedDomain, we said we will restrict to 1. If DID Document has 1 Linked Domain type and a different type service, this case will fail.

Reply:
Fixed.